app/controllers/carts_controller.rb

```diff
@@ -4,7 +4,16 @@ class CartsController < ApplicationController
   before_action :check_user_cart
 
   def add_product
-    line_item = @cart.line_items.build(product: @product)
+    line_item = @cart.line_items.find_by(product: @product)
```

Comment:
Can you please also explain what you are trying to achieve in this function

Reply:
This action is for adding a product to a cart and if a user clicks again on the add to cart button for the same product then one of the conditions triggers and it just increases product quantity. It doesn't add a new product to a cart. Otherwise, for different products, it adds a new product.